### PR TITLE
ls improvements

### PR DESCRIPTION
**Pull request by teknoraver**
added two ls argument:
```
  -a    show hidden files
  -h    human readable sizes
```

### CODE REVIEW

File: cmds/ls/fileinfo_linux.go [hunk 1]
```diff
@@ -106,14 +107,33 @@ type quotedStringer struct {
...
+	comp  fmt.Stringer // decorator pattern
+	human bool
 }
 
 // Return the name surrounded by quotes with escaped control characters.
 func (fi quotedStringer) String() string {
 	return fmt.Sprintf("%#v", fi.name)
 }
 
+func sizeToString(size int64) string {
```
Comment (hugelgupf) [inline cmds/ls/fileinfo_linux.go]: This looks good to me, but what about something like https://github.com/dustin/go-humanize
?

File: cmds/ls/fileinfo_linux.go [hunk 2]
```diff
@@ -106,14 +107,33 @@ type quotedStringer struct {
...
+	switch {
```
Comment (rminnich) [inline cmds/ls/fileinfo_linux.go]: This needs to be in a package. I am not that concerned about what package, but having it special in ls is taking us down a bad path.

File: cmds/ls/ls.go [hunk 3]
```diff
@@ -67,8 +73,11 @@ func listName(d string, w io.Writer, prefix bool) error {
 			}
 		}
 
-		// Print the file in the proper format.
-		fmt.Fprintln(w, stringer(fi))
+		// Hide .files unless -a was given
```
Comment (rminnich) [inline cmds/ls/ls.go]: . file issues should be a separate PR. Not all agree that hiding . files was a good idea.
https://plus.google.com/+RobPikeTheHuman/posts/R58WgWwN9jp
The question, should we continue this mistake? Is there harm in just printing .files?I don't see a problem.

**Comment by rminnich** [comment]
Do these packages have anything you can use?

from cbmem:
// The C version of cbmem has a complex function to list
// numbers in xx,xxx,xxx form. I personally think this
// is a terrible idea (what about the EU among other things?)
// If you decide yor really want this, then don't write a function,
// do this.
//    "golang.org/x/text/language"
//    "golang.org/x/text/message"
//    p := message.NewPrinter(language.English)
//    p.Printf("%d\n", 1000)

**Comment by hugelgupf** [comment]
-a is normal expected Linux behavior. Humans and scripts rely on it. We're gonna have to implement it.

Let's make use of that humanize package.

File: cmds/ls/ls.go (hunk 3 above)
Comment (hugelgupf) [reply cmds/ls/ls.go]: This PR has two commits to separate this. I don't think a separate PR is necessary.

**Comment by teknoraver** [comment]
@hugelgupf I used the humanize package in the second PR

**Comment by hugelgupf** [comment]
hey -- it seems you added go-humanize to the vendor config, but not actually to the tree.

to do that, please run:
```
$ go get -u github.com/golang/dep/cmd/dep
$ dep ensure
$ dep prune
```

**Comment by rminnich** [comment]
Sorry to drag this on so long, do you want to finish it? Thanks.

**Comment by teknoraver** [comment]
@rminnich I don't know how to fix the missing dependency.
I'm running `dep ensure` and committing Gopkg.lock, am I missing something?

**Comment by bensallen** [comment]
@teknoraver Make sure you commit the changes in ./vendor as well. Also please ensure you run `dep prune` after running `dep ensure`.

**Comment by teknoraver** [comment]
@rminnich I've commited ./vendor and updated the ls test to comply with the new output

**Comment by rminnich** [comment]
are we good to go here?
